docs/docs/tutorials/orca-yarn-tutorial.md

```diff
@@ -0,0 +1,475 @@
+``` bash
+# "bigdl" is conda environment name, you can use any name you like.
+# Please change Python version to 3.6 if you need a Python 3.6 environment.
+conda create -n bigdl python=3.7 
+conda activate bigdl
+```
+Alternatively, You can install the latest release version of BigDL (built on top of Spark 2.4.6 by default) as follows:
+```bash
+pip install bigdl
+```
+You can install the latest nightly build of BigDL as follows:
+```bash
+pip install --pre --upgrade bigdl
+```
+
+__Note:__ 
+* Using Conda to install BigDL will automatically install libraries including `conda-pack`, `torch`, `torchmetrics`, `torchvision`, `pandas`, `numpy`, `pyspark==2.4.6`, and etc.
+* You can install BigDL built on top of Spark 3.1.2 as follows:
+    
+    ```bash
+    # Install the latest release version
+    pip install bigdl-spark3
+
+    # Install the latest nightly build version
+    pip install --pre --upgrade bigdl-spark3 
+    ```
+* Installing bigdl-spark3 will automatically install `pyspark==3.1.2`.
+
+Please see more details in [Python User Guide](https://bigdl.readthedocs.io/en/latest/doc/UserGuide/python.html).
+
+## 2.3 For CDH Users
+If the __client node__ where you submit applications has installed Spark, it might conflict with the `pyspark` installed automatically with BigDL on the __client node__ if they versions not match. Please follow the steps below to solve the conflict.
+
+On the __client node__ that you submit applications, please unset Spark related environment variables to avoid the conflict.
+```bash
+unset SPARK_HOME
+unset SPARK_VERSION
+```
+You could also use `env | grep SPARK` to find all the existing Spark environment variables, please usnet them all.
+
+__Note:__
+* For CDH cluster user, the `HADOOP_CONF_DIR` should be `/etc/hadoop/conf` on CDH by default.
+* If you are using `pyarrow` to connnect with network file systems, please setup `ARROW_LIBHDFS_DIR` on the __client node__ where you submit applications.
+    ```bash
+    locate libhdfs.so
+    ```
+    You will get a list locations of `libhdfs.so`, please assign the directory path used in CDH to the `ARROW_LIBHDFS_DIR` variable.
+    ```bash
+    export ARROW_LIBHDFS_DIR=/opt/cloudera/parcels/CDH-${CHD_VERSION}/lib64/
+    ```
+    Please see __Part5__ to see more details for running the example.
+
+# 3. Prepare Dataset 
+To run the example on YARN, you should upload the Fashion-MNIST dataset to network file systems (HDFS or S3).   
+
+First, please download the Fashion-MNIST dataset manually on your __client node__.
+```bash
+# PyTorch official dataset download link
+git clone https://github.com/zalandoresearch/fashion-mnist.git
+
+mv /path/to/fashion-mnist/data/fashion /path/to/local/data/FashionMNIST/raw 
+```
+Then upload it to the network file system.
+```bash
+# Upload to HDFS
+hdfs dfs -put /path/to/local/data/FashionMNIST hdfs://url:port/path/to/remote/data
+```
+
+# 4. Prepare Custom Module
```

Comment:
This is probably only needed in yarn-cluster mode; if so, move this section to the Yarn-Cluster mode section

Reply:
This section is still needed in `yarn-client` mode, have updated to using `from model import ...` instead to keep smoothly between yarn and local.